example-app/src/main/java/backtraceio/backtraceio/MainActivity.java

```diff
@@ -112,48 +114,51 @@ private BacktraceClient initializeBacktrace(final String submissionUrl) {
     public native void cppCrash();
 
     public native boolean registerNativeBreadcrumbs(BacktraceBase backtraceBase);
+
     public native boolean addNativeBreadcrumb();
+
     public native boolean addNativeBreadcrumbUserError();
+
     public native void cleanupNativeBreadcrumbHandler();
 
     private List<String> equippedItems;
 
-    public List<String> getWarriorArmor()
-    {
+    public List<String> getWarriorArmor() {
         return new ArrayList<String>(Arrays.asList("Tough Boots", "Strong Sword", "Sturdy Shield", "Magic Wand"));
     }
 
-    int findEquipmentIndex(List<String> armor, String equipment)
-    {
+    int findEquipmentIndex(List<String> armor, String equipment) {
         return armor.indexOf(equipment);
     }
 
-    void removeEquipment(List<String> armor, int index)
-    {
+    void removeEquipment(List<String> armor, int index) {
         armor.remove(index);
     }
 
-    void equipItem(List<String> armor, int index)
-    {
+    void equipItem(List<String> armor, int index) {
         equippedItems.add(armor.get(index));
     }
 
     public void handledException(View view) {
         try {
-            List<String> myWarriorArmor = getWarriorArmor();
-            int magicWandIndex = findEquipmentIndex(myWarriorArmor, "Magic Wand");
-            // I don't need a Magic Wand, I am a warrior
-            removeEquipment(myWarriorArmor, magicWandIndex);
-            // Where was that magic wand again?
-            equipItem(myWarriorArmor, magicWandIndex);
-        } catch (Exception e) {
-            backtraceClient.send(new BacktraceReport(e));
+            try {
+                List<String> myWarriorArmor = getWarriorArmor();
+                int magicWandIndex = findEquipmentIndex(myWarriorArmor, "Magic Wand");
+                // I don't need a Magic Wand, I am a warrior
+                removeEquipment(myWarriorArmor, magicWandIndex);
+                // Where was that magic wand again?
+                equipItem(myWarriorArmor, magicWandIndex);
+            } catch (IndexOutOfBoundsException e) {
+                throw new IndexOutOfBoundsException("Invalid index of selected element!");
+            }
+        } catch (IndexOutOfBoundsException e) {
+            backtraceClient.send(new BacktraceReport(e), this.listener);
```

Comment:
wait, why? I see we want to show inner errors case here, however, can we make it more prominent?

Reply:
it's bugfix for integration test purpose.

in shortcut: different SDKs/devices contains different default message in IndexOutOfBoundsException so if we want to verify in the integration test if certain exception was thrown and we would like to verify error message we need to have it always in the same form and with the same value 

<img width="614" alt="image" src="https://github.com/backtrace-labs/backtrace-android/assets/22329621/d73eee52-48f2-4a7f-87c0-521d3e1d2958">

If you want I can keep it as it is or try to rearrange the method to e.g. throw another type of exception which will not cause problems like this